mock/world/worldUpdate.go

```diff
@@ -58,14 +58,16 @@ func (b *MockWorld) UpdateAccounts(
 		acct := b.AcctMap.GetAccount(modAcct.Address)
 		if acct == nil {
 			acct = &Account{
-				Exists:       false,
-				Address:      modAcct.Address,
-				Nonce:        0,
-				Balance:      zero,
-				Storage:      make(map[string][]byte),
-				Code:         nil,
-				OwnerAddress: modAcct.CodeDeployerAddress,
-				ESDTData:     make(map[string]*ESDTData),
+				Exists:          false,
+				Address:         modAcct.Address,
+				Nonce:           0,
+				Balance:         zero,
+				Storage:         make(map[string][]byte),
+				Code:            nil,
+				OwnerAddress:    modAcct.CodeDeployerAddress,
+				ESDTData:        make(map[string]*ESDTData),
+				ShardID:         0,
+				IsSmartContract: false,
 			}
 			b.AcctMap.PutAccount(acct)
 		}
```

Comment:
All the `if`s below could be added in a method of the `mock.Account` struct. Something like `acct.UpdateFromOutputAccount(modAcct)`.